src/lib/Libwin/passwd.c

```diff
@@ -4816,18 +4816,20 @@ logon_pw(char *username,
 	int use_winsta,
 	char msg[4096])
 {
-	static  char	*id = "logon_pw";
-	int		i;
-	struct	passwd *pwdp;
-	SID	*usid;
-	char	*homedir;
-	char  	thepass[PWLEN+1];
-	char 	realname[UNLEN+1];
-	char	domain[PBS_MAXHOSTNAME+1] = "";
-	char	*p;
-	char	msg2[4096];
+	static const char __func__[] = "logon_pw";
+	int		i = 0;
+	struct	passwd *pwdp; = NULL
+	SID	*usid = NULL;
+	char	*homedir = NULL;
+	char  	thepass[PWLEN+1] = {'\0'};
+	char 	realname[UNLEN+1] = {'\0'};
+	char	domain[PBS_MAXHOSTNAME+1] = {'\0'};
+	char	*p = {'\0'};
+	char	msg2[4096] = {'\0'};
```

Comment:
It is a good practice to define a macro that indicates the array size. In this case, you have two character arrays that you want to make the same size. Using a macro will ensure they stay consistent.